internal/pkg/istio/config_factory.go

```diff
@@ -199,74 +286,74 @@ func (cf *ConfigFactory) GetWorkloadEntries() ([]*v1alpha3.WorkloadEntry, error)
 	return workloadEntries, nil
 }
 
-func (cf *ConfigFactory) GetVirtualServices() *v1alpha3.VirtualService {
-	return &v1alpha3.VirtualService{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      federationIngressGatewayName,
-			Namespace: cf.cfg.MeshPeers.Local.ControlPlane.Namespace,
-		},
-		Spec: istionetv1alpha3.VirtualService{
-			Hosts:    []string{"*"},
-			Gateways: []string{federationIngressGatewayName},
-			Tcp: []*istionetv1alpha3.TCPRoute{{
-				Match: []*istionetv1alpha3.L4MatchAttributes{{
-					Port: cf.cfg.MeshPeers.Local.Gateways.Ingress.Ports.GetDiscoveryPort(),
-				}},
-				Route: []*istionetv1alpha3.RouteDestination{{
-					Destination: &istionetv1alpha3.Destination{
-						Host: cf.controllerServiceFQDN,
-						Port: &istionetv1alpha3.PortSelector{
-							Number: 15080,
-						},
-					},
-				}},
-			}},
-		},
-	}
-}
-
-func (cf *ConfigFactory) getServiceEntryForRemoteFederationController() *v1alpha3.ServiceEntry {
-	if len(cf.cfg.MeshPeers.Remote.Addresses) == 0 {
-		return nil
-	}
-
-	var endpoints []*istionetv1alpha3.WorkloadEntry
-	for _, remoteAddr := range cf.cfg.MeshPeers.Remote.Addresses {
-		endpoints = append(endpoints, &istionetv1alpha3.WorkloadEntry{Address: remoteAddr})
-	}
-	return &v1alpha3.ServiceEntry{
+func (cf *ConfigFactory) serviceEntryForRemoteFederationController() *v1alpha3.ServiceEntry {
+	se := &v1alpha3.ServiceEntry{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "remote-federation-controller",
 			Namespace: cf.cfg.MeshPeers.Local.ControlPlane.Namespace,
 			Labels:    map[string]string{"federation.istio-ecosystem.io/peer": "todo"},
 		},
-		Spec: istionetv1alpha3.ServiceEntry{
-			Hosts: []string{fmt.Sprintf("remote-federation-controller.%s.svc.cluster.local", cf.cfg.MeshPeers.Local.ControlPlane.Namespace)},
+	}
+	if cf.cfg.MeshPeers.Remote.IngressType == config.OpenShiftRouter {
+		se.Spec = istionetv1alpha3.ServiceEntry{
+			Hosts:     []string{cf.cfg.MeshPeers.Remote.Addresses[0]},
+			Addresses: common.Resolve(cf.cfg.MeshPeers.Remote.Addresses[0]),
 			Ports: []*istionetv1alpha3.ServicePort{{
-				Name:     "discovery",
+				Name:     "tls-passthrough",
+				Number:   cf.cfg.MeshPeers.Remote.GetPort(),
+				Protocol: "TLS",
+			}},
+			Location:   istionetv1alpha3.ServiceEntry_MESH_INTERNAL,
+			Resolution: istionetv1alpha3.ServiceEntry_DNS,
+		}
+	} else {
+		se.Spec = istionetv1alpha3.ServiceEntry{
+			// TODO: this will not work for ingressType=istio when the remote address is a hostname
+			Hosts:     []string{fmt.Sprintf("federation-discovery-service-%s.istio-system.svc.cluster.local", cf.cfg.MeshPeers.Remote.Name)},
+			Addresses: common.Resolve(cf.cfg.MeshPeers.Remote.Addresses[0]),
```

Comment:
Would this make sense?
```suggestion
			Addresses: slices.Flatten(slices.Map(cf.cfg.MeshPeers.Remote.Addresses, func(addr string) []string {
					return common.Resolve(addr)
			})),
```

Reply:
In general more than one address does not make any sense, because all of them will be resolved to the same set of IPs.